src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -1060,14 +1072,15 @@ void VM_Version::get_processor_features() {
   }
 
     // Currently APX support is only enabled for targets supporting AVX512VL feature.
-  bool apx_supported = os_supports_apx_egprs() && supports_apx_f() && supports_avx512vl();
+  bool apx_supported = os_supports_apx_egprs() && supports_apx_f() && supports_apx_nci_ndd_nf() && supports_avx512vl();
   if (UseAPX && !apx_supported) {
     warning("UseAPX is not supported on this CPU, setting it to false");
     FLAG_SET_DEFAULT(UseAPX, false);
   }
 
   if (!UseAPX) {
     _features.clear_feature(CPU_APX_F);
+    _features.clear_feature(CPU_APX_NCI_NDD_NF);
```

Comment:
We don't need separate CPU_APX_NCI_NDD_NF feature and the related changes as CPU_APX_F feature is set only when both bits (sefsl1_cpuid7_edx.bits.apx_f and std_cpuid29_ebx.bits.apx_nci_ndd_nf) are set.

Reply:
Please see the updated changes which removes CPU_APX_NCI_NDD_NF as an explicit feature.